azurerm/resource_arm_hd_insight_clusters.go

```diff
@@ -0,0 +1,918 @@
+				ValidateFunc: validation.StringInSlice([]string{
+					"Linux",
+					"Windows",
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+			"resource_group_name": resourceGroupNameSchema(),
+			"restauth_username": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
+			"restauth_password": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
+			"head_node": {
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"linux_os_profile": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"password": {
+										Optional: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"ssh_keys": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeList,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"key_data": {
+													Required: true,
+													ForceNew: true,
+													Type:     schema.TypeString,
+												},
+											},
+										},
+									},
+									"username": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+								},
+							},
+						},
+						"min_instance_count": {
+							Optional:     true,
+							Default:      1,
+							ForceNew:     true,
+							Type:         schema.TypeInt,
+							ValidateFunc: validation.IntBetween(1, 32),
+						},
+						"script_actions": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeList,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"parameters": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"uri": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+								},
+							},
+						},
+						"target_instance_count": {
+							Required:     true,
+							ForceNew:     true,
+							Type:         schema.TypeInt,
+							ValidateFunc: validation.IntBetween(1, 32),
+						},
+						"vm_size": {
+							Required: true,
+							ForceNew: true,
+							Type:     schema.TypeString,
+						},
+						"vnet_profile": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"subnet": {
```

Comment:
rather than requiring the Subnet Name and the Virtual Network ID - let's instead accept a Subnet ID and parse this information out of the URI